nilearn/_utils/param_validation.py

```diff
@@ -0,0 +1,65 @@
+"""
+Utilities to check for valid parameters
+"""
+
+import numbers
+import warnings
+
+from .compat import _basestring
+
+
+def check_threshold(threshold, data, percentile_calculate, name='threshold'):
+    """ Checks if the given threshold is in correct format and within the limit.
+
+    If necessary, this function also returns score of the data calculated based
+    upon the given specific percentile function.
+    Note: This is only for threshold as string.
+
+    Parameters
+    ----------
+    threshold: a float value or a real number or a percentage in string.
```

Comment:
What difference do you have in mind between float and real number? The first line is to indicate types, you should be as terse as possible, e.g. float or str. More details can be specified in the section below.